src/TranslationResult.h

```diff
@@ -8,101 +8,178 @@
 #ifndef SRC_TRANSLATOR_TRANSLATIONRESULT_H_
 #define SRC_TRANSLATOR_TRANSLATIONRESULT_H_
 
+#include <cassert>
 #include <string>
+#include <utility>
 #include <vector>
 
 #include "QualityScore.h"
 
 /* This class represents the result of AbstractTranslationModel::translate() API
  * for each of its text entry and TranslationRequest.
  */
+
+#include "translator/sentence_ranges.h"
+
+/// Annotation is a pseudo-2D container which holds a sequence of
+/// string_views that correspond to words in vocabulary from either source or
+/// the translation. The API allows adding a sentence, which accounts extra
+/// extra annotation to mark sentences in a flat vector<string_views> storing
+/// them in an efficient way.
+
+/// With Annotation construct in place, we bind an std::string and string_views
+/// that are valid referring to the std::string in an AnnotatedBlob.
+/// AnnotatedBlob doesn't have restricting it to marian::bergamot only. For DRY
+/// for development and debugging purposes, it's typedef-ed using a structure
+/// currently marian internal.
+typedef marian::bergamot::AnnotatedBlobT<std::string_view> AnnotatedBlob;
+
+/// Alignment is stored as a sparse matrix, this pretty much aligns with marian
+/// internals but is brought here to maintain translator
+/// agnosticism/independence.
+struct Point {
+  size_t src; // Index pointing to source string_view.
+  size_t tgt; // Index pointing to target string_view.
+  float prob; // Score between [0, 1] on indicating degree of alignment.
+};
+
+typedef std::vector<Point> Alignment;
+
+struct Quality {
+  float sequence; /// Certainty/uncertainty score for sequence.
+  /// Certainty/uncertainty for each word in the sequence.
+  std::vector<float> word;
```

Comment:
Few questions regarding this struct:

1. What does `sequence` and `word` data member represent?
2. How will the `TranslationResult` class return Quality Scores at [2 granularity levels](https://github.com/browsermt/bergamot-translator/blob/jp/word-alignments/src/QualityScore.h#L15-L16)?

Reply:
sequence is the sequence-level quality. vector<float> is the word-level quality, empty if there's no word level quality.

Reply:
What is sequence-level quality?

How will the TranslationResult class return Quality Scores at 2 granularity levels?

> Few questions regarding this struct:
> 
> sequence is the sequence-level quality.

What is a sequence-level quality?

>     2. How will the `TranslationResult` class return Quality Scores at [2 granularity levels](https://github.com/browsermt/bergamot-translator/blob/jp/word-alignments/src/QualityScore.h#L15-L16)?

Could you answer this one?

Reply:
They're logprobs (not great values, but works to prototype).  At the moment:

```
sequence = -sum(log(prob-word[i]))
word[i]  = log(prob-word[i])
```

It is not always that sequence scores are going to be a reduction from word scores. We can probably use an ensemble's entropy, entropy in output classes etc as proxy for quality scores alternatively. Maybe even another network predicting quality. However, since I have some background in the field, I can assure something covered by this struct is what you'll always get from any translation framework. 

> Could you answer this one?

Haven't I already? sequence => sentence granularity, vector<float> for words => word level granularity, empty if not requested. Would you like a rename or something?

Reply:
> Haven't I already? sequence => sentence granularity

This wasn't clear above. Thanks for clarification 👍